impeller/docs/standalone_gles.md

```diff
@@ -0,0 +1,195 @@
+
+  // |ReactorGLES::Worker|
+  bool CanReactorReactOnCurrentThreadNow(
+      const ReactorGLES& reactor) const override {
+    return true;
+  }
+};
+```
+
+Add an instance of this reactor worker to the context. Whew, that was unnecessarily complicated. But this is only necessary for OpenGL ES because of its tricky threading. Skip this step for Vulkan and Metal. Those APIs are already thread safe.
+
+```c++
+context->AddReactorWorker(worker);
+```
+
+Once you have the context, use it to crate a renderer.
+
+```c++
+auto renderer = std::make_shared<Renderer>(context);
+```
+
+And setup is done. Keep the context and renderer around. We will be using it during frame rendering.
+
+# Rendering Frames
+
+Rendering frames is a matter of:
+* Wrapping the onscreen texture as a Surface (`impeller::SurfaceGLES` in our case).
+* Giving the surface to our renderer to render.
+* The renderer will prepare a render target from this surface and invoke a callback that we'll use to setup a render pass directed at that render target.
+* Populate the render pass how we see fit.
+* Tell our reactor worker that all operations need to be flushed to OpenGL.
+* Present the onscreen surface.
+* Repeat.
+
+## Wrap the Onscreen Surface
+
+Per frame, the onscreen surface can be wrapped using SurfaceGLES::WrapFBO where the default framebuffer in our case if FBO 0. Take care to to ensure that the pixel format matches the one we used to choose the EGL config. Figuring out the pixel size is left as an exercise for the reader.
+
+```c++
+auto surface =
+    SurfaceGLES::WrapFBO(context,                         // context
+                         swap_callback,                   // swap callback
+                         0u,                              // fbo
+                         PixelFormat::kR8G8B8A8UNormInt,  // pixel format
+                         GetWindowSize()                  // surface size
+    );
+```
+
+## Setup the Swap Callback
+
+The swap callback will get invoked when the renderer presents the surface. Remember in our list of things to do, we need to first tell the reactor worker to flush all pending OpenGL operations and then present the surface. Setup the swap callback appropriately.
+
+```c++
+SurfaceGLES::SwapCallback swap_callback =
+    [surface, context]() -> bool {
+  context->GetReactor()->React();
+  return surface->Present();
+};
+```
+
+## Render to the Surface
+
+Give the surface to the rendere along with a callback that details how you will populate the render target the renderer sets up that is directed at that surface.
+
+```c++
+renderer_->Render(std::move(surface),
+                 [&](RenderTarget& render_target) -> bool {
+                    // Do things that render into the render target.
+                    return true;
+                 });
+```
+
+And that's it. Now you have functional WSI and render loop. Higher level frameworks like Aiks and DisplayList that use the render target to render their rendering into to the surface.
```

Comment:
So if I'm understanding this correctly, I would be able to create an `AiksContext` allowing me to use the Aiks API?

Reply:
Yes. Aiks also needs a typographer to layout and render text. Most layers in the API stack have a context and you build one using the context of the layer below. Feel free to file a bug for documentation on how to set that up (or any other topic).

Reply:
Cheers!